x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/LikeEscapingParsingTests.java

```diff
@@ -63,8 +63,15 @@ public void testInvalidChar() {
                 is("line 1:28: Char [%] cannot be used for escaping"));
     }
 
-    public void testCannotUseStar() {
+    public void testCanUseStarWithoutEscaping() {
+        LikePattern like = like("%string*");
+        assertThat(like.pattern(), is("%string*"));
+        assertThat(like.asJavaRegex(), is("^.*string\\*$"));
+        assertThat(like.asLuceneWildcard(), is("*string\\*"));
+    }
+
+    public void testCannotUseStarWithEscaping() {
         assertThat(error("'|*string' ESCAPE '|'"),
-                is("line 1:11: Invalid char [*] found in pattern [|*string] at position 1; use [%] or [_] instead"));
+            is("line 1:11: Pattern [|*string] is invalid as escape char [|] at position 0 can only escape wildcard chars; found [*]"));
```

Comment:
Should we maybe improve the error message by listing the wildcard chars for LIKE: `%` and `_` ?

Reply:
> Looks good, Maybe add one more test with a more complicated string, something like:
> `*%*_string*_**%`. Left one more comment about the error message.

Will add a new test case, but I will add it to the [LikeConversionTests](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/util/LikeConversionTests.java), since I think that has less to do with `ESCAPE`ing.